diff --git a/CMakeLists.txt b/CMakeLists.txt
index 839e2b6..46c0573 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,9 +6,9 @@ find_package(catkin REQUIRED COMPONENTS rosconsole sensor_msgs)
 if(NOT ANDROID)
   find_package(PythonLibs)
 
-  if(PYTHONLIBS_VERSION_STRING VERSION_LESS "3.8")
+  if(PYTHONLIBS_VERSION_STRING VERSION_LESS "3.9")
     # Debian Buster
-    find_package(Boost REQUIRED python37)
+    find_package(Boost REQUIRED python38)
   else()
     # Ubuntu Focal
     find_package(Boost REQUIRED python)
